---
title: "GCTA"
subtitle: "a tool for Genome-wide Complex Trait Analysis"
author: "王　超辰"
date: "2017/5/16"
output:
  xaringan::moon_reader:
    css: zh-CN.css
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
```

## Estimating the Variance Explained by Genome-Wide SNPs

### Basic concept:
- To fit all the SNPs simultaneously as **random effects** in a mixed linear model.

- To estimate **the variance explained by all the SNPs** without testing for associations of any individual SNP with the trait.




---

## Model (1)

$$y = Xb + Wu + e \\
with\\ var(y)=V=WW^{\prime}\sigma_u^2+I\sigma_e^2 \;\;(1)$$

- $y$ is a $n \times 1$ vector of phenotypes(身長，BMI，糖尿病ありなしetc.);<br> $n =$ sample size;
- $b$ is a vector of fixed covariates.
- $u$ is a vector of **SNP effects** with $u \sim N(0, I\sigma_u^2)$
- $I$ is a $n \times n$ identity matrix (単位行列)
- $e$ is a vector of residual effects with $e \sim N(0, I\sigma_e^2)$
- $W$ is a standardized SNP genotype matrix with the $ij$th element $w_{ij}=\frac{(x_{ij}-2p_i)}{\sqrt{2p_i(1-p_i)}}$, where $x_{ij}$ is the number of copies of the reference allele for the $i$th SNP of the $j$th individual and $p_i$ is the frequency of the reference allele of SNP $i$.

---
## Model (2)

#### Define $A=\frac{WW^{\prime}}{N}$
#### Define $\sigma_G^2$ as the variance explained by all the SNPs i.e., <br> $\sigma_G^2 = N\sigma_u^2$ with $N=$ number of SNPs.
#### Equation (1) $$y = Xb + Wu + e \\ with\\ var(y)=V=WW^{\prime}\sigma_u^2+I\sigma_e^2 \;\;(1)$$
#### is equivalent to equation (2) $$y = Xb + g + \epsilon \\ with \\ var(y)=V=A\sigma_G^2+I\sigma_e^2 \;\;(2)$$



---
class: middle
## Equation (2)
### $$y = Xb + g + \epsilon \\ with \\ var(y)=V=A\sigma_G^2+I\sigma_e^2 \;\;(2)$$

- $g$ is a $n \times 1$ vector of the total additive genetic effects of the individuals with $g\sim N(0, A\sigma_G^2)$
- $A=$ **genetic relationship matrix (GRM)** between individuals with the $kj$th element being $A_{kj}=\frac{1}{N}\sum_{i-1}^N\frac{(x_{ij}-2p_i)(x_{ik}-2p_i)}{2p_i(1-p_i)}$
- $\sigma_G^2$ and $\sigma_e^2$ can be estimated by restricted maximum likelihood (REML) approach.


---
### $h_{SNP}^2$

- **Define** $h_{SNP}^2=\frac{\sigma_G^2}{(\sigma_G^2+\sigma_e^2)}$ as the **proportion of phenotypic variance explained by all the SNPs**;
- $h_{SNP}^2$ is **directly comparable** to the results from GWAS analyses.
- **For Example, Human Height:**
    - $\sim 10\%$ of variance explained by $180$ genome-wide significant SNPs $(p<5\times 10^{-8})$<sup>[1]</sup>;
    - $h_{SNP}^2=0.45 (s.e.=0.08)$, meaning all common SNPs explain $\sim 45\%$ of the phenotypic variance<sup>[2]</sup>
- Distinction needed: $h_{SNP}^2\neq (h^2) heritability$
    - current generation of genotyping arrays are unlikely to perfectly tag all causal variants for a trait;
    - $h_{SNP}^2$ from unrelated individuals is **an unbiased estimate of $h^2$**.

.footnote[
[1] Lango Allen, et al. (2010). Hundreds of variants clustered in genomic loci and biological pathways affect human height. Nature 467, 832–838.

[2] Yang, J., et al. (2010). Common SNPs explain a large proportion of the heritability for human height. Nat. Genet. 42, 565–569.
]


---

## [GCTA applications:](http://cnsgenomics.com/software/gcta/index.html)

* Step 1: generating **genetic relationship matrix (GRM)**<br>

    + a GWAS dataset in PLINK binary PED format (`test.bed; test.bim; test.fam`)
    + _Command 1_:
```yaml
./gcta64 --bfile test --autosome --make-grm --out test
```
    + _Command 2_: GRM for each chromosome (large dataset)
```yaml
./gcta64 --bfile test --chr1 --make-grm --out test_chr1
./gcta64 --bfile test --chr2 --make-grm --out test_chr2
...
./gcta64 --bfile test --chr22 --make-grm --out test_chr22
```
    + and then merge all the GRMs by _command 3_:
```yaml
./gcta64 --mgrm grm_chrs.txt --make-grm --out test
```

---
* Step 2: estimate the variance explained by all SNPs by a Restricted Maximum Likelihood (REML) analysis of the phenotypes with the GRM.


---
## Extension: Genetic correlation between two traits

> Comparing the phenotypic similarity and the genetic similarity between individuals **within** and **cross** two traits.


$$cov(y_{i1},y_{k1})=A_{ik}\sigma_{g1}^2\\var(y_{i1})=A_{ii}\sigma_{g1}^2+\sigma_{e1}^2$$
$$cov(y_{i2},y_{k2})=A_{ik}\sigma_{g2}^2\\var(y_{i2})=A_{ii}\sigma_{g2}^2+\sigma_{e2}^2$$
$$cov(y_{i1},y_{k2})=A_{ik}\sigma_{g1g2}\\cov(y_{i1},y_{i2})=A_{ii}\sigma_{g1g2}+\sigma_{e1e2}$$

"1" = trait 1, "2" = trait 2; $\sigma_{g1g2} =$ genetic covariance; $\sigma_{e1e2}=$ environmental covariance<br>
$${\bf Genetic\;correlation}\rightarrow r_g=\frac{\sigma_{g1g2}}{\sigma_{g1}\sigma_{g2}}$$
